### PR TITLE
Create LedgerRpcProvider. Implement RPC slot RPC

### PR DESCRIPTION
**Pull request by preston-evans98**

### CODE REVIEW

File: db/sovereign-db/src/ledger_db/mod.rs [hunk 1]
```diff
@@ -118,6 +124,64 @@ impl<S: SlotData> LedgerDB<S> {
...
+        range: &std::ops::Range<TxNumber>,
+    ) -> Result<Vec<StoredTransaction>, anyhow::Error> {
+        self.get_data_range::<TxByNumber, _, _>(range)
+    }
+
+    /// Gets all data with identifier in `range.start` to `range.end`. If `range.end` is outside
+    /// the range of the database, the result will smaller than the requested range.
+    /// Note that this method blindly preallocates for the requested range, so it should not be exposed
+    /// directly via rpc.
+    fn get_data_range<T: Schema<Key = K, Value = V>, K: Into<u64> + Copy + SeekKeyEncoder<T>, V>(
```
Comment (bkolad) [inline db/sovereign-db/src/ledger_db/mod.rs:164]: Let's use `where` clause here (I think it is more readable)

File: db/sovereign-db/src/ledger_db/mod.rs [hunk 2]
```diff
@@ -118,6 +124,64 @@ impl<S: SlotData> LedgerDB<S> {
...
+        &self,
+        range: &std::ops::Range<K>,
+    ) -> Result<Vec<V>, anyhow::Error> {
+        let mut raw_iter = self.db.iter()?;
+        let max_items = (range.start.into() - range.end.into()) as usize;
+        raw_iter.seek(&range.start)?;
+        let mut iter = raw_iter.take(max_items);
+        let mut out = Vec::with_capacity(max_items);
+        for _ in 0..max_items {
```
Comment (bkolad) [inline db/sovereign-db/src/ledger_db/mod.rs:173]: here we can just iterate over `iter ` (and remove the pattern matching)

**Review by bkolad — APPROVED** [review]

File: db/sovereign-db/src/ledger_db/mod.rs (hunk 1 above)
Comment (preston-evans98) [reply db/sovereign-db/src/ledger_db/mod.rs:164]: Done!

File: db/sovereign-db/src/ledger_db/mod.rs (hunk 2 above)
Comment (preston-evans98) [reply db/sovereign-db/src/ledger_db/mod.rs:173]: Good thinking. Fixed!

**Pull request merged**